libethereum/State.cpp

```diff
@@ -271,7 +271,9 @@ u256 State::balance(Address const& _id) const
 
 void State::incNonce(Address const& _addr)
 {
-	if (Account* a = account(_addr))
+	if (_addr == MaxAddress)
```

Comment:
I would better put this special logic into `Executive`, because it's more "the logic of executing special kind of transactions" than "the logic of incrementing the nonce in the state", and it would be closer to other special treatment of zero sig transactions.

If you want to avoid duplication in several places where `incNonce` is called, make one wrapper method with this check